cross-media-measurement/src/test/kotlin/org/wfanet/measurement/common/grpc/TransportSecurityTest.kt

```diff
@@ -0,0 +1,269 @@
+      client.check(
+        HealthCheckRequest.newBuilder().apply {
+          service = SERVICE
+        }.build()
+      )
+    }
+
+    assertThat(response.status).isEqualTo(HealthCheckResponse.ServingStatus.SERVING)
+  }
+
+  @Test
+  fun `mTLS RPC succeeds`() {
+    val server = startServer(ClientAuth.REQUIRE)
+    val channel = grpcCleanup.register(
+      NettyChannelBuilder.forAddress(HOSTNAME, server.port)
+        .sslContext(clientCerts.toClientTlsContext())
+        .build()
+    )
+    val client = HealthCoroutineStub(channel)
+
+    val response = runBlocking {
+      client.check(
+        HealthCheckRequest.newBuilder().apply {
+          service = SERVICE
+        }.build()
+      )
+    }
+
+    assertThat(response.status).isEqualTo(HealthCheckResponse.ServingStatus.SERVING)
+  }
+
+  companion object {
+    val logger: Logger = Logger.getLogger(this::class.java.canonicalName)
+
+    @JvmField
+    @ClassRule
+    val temporaryFolder: TemporaryFolder = TemporaryFolder()
+
+    @JvmStatic
+    @BeforeClass
+    fun generateCerts() = runBlocking {
+      val extFile = temporaryFolder.root.resolve("san.ext")
+      extFile.writeText(SUBJECT_ALT_NAME_EXT)
+
+      generateRootCert("server-root", "/O=Server CA/CN=ca.server.example.com")
+      generateSigningRequest("server", "/O=Server/CN=server.example.com")
+      signCertificate("server", "server-root")
+
+      generateRootCert("client-root", "/O=Client CA/CN=ca.client.example.com")
+      generateSigningRequest("client", "/O=Client/CN=client.example.com")
+      signCertificate("client", "client-root")
+    }
+
+    private suspend fun generateRootCert(name: String, subject: String) {
+      runCommand(
+        "openssl",
+        "req",
+        "-out",
+        "$name.pem",
+        "-new",
+        "-newkey",
+        ALGORITHM,
+        "-pkeyopt",
+        "ec_paramgen_curve:$CURVE",
+        "-nodes",
+        "-keyout",
+        "$name.key",
+        "-x509",
+        "-addext",
+        SUBJECT_ALT_NAME_EXT,
+        "-days",
+        "3650",
+        "-subj",
+        subject
+      )
+    }
+
+    private suspend fun generateSigningRequest(name: String, subject: String) {
+      runCommand(
+        "openssl",
+        "req",
+        "-out",
+        "$name.csr",
+        "-new",
+        "-newkey",
+        ALGORITHM,
+        "-pkeyopt",
+        "ec_paramgen_curve:$CURVE",
+        "-nodes",
+        "-keyout",
+        "$name.key",
+        "-addext",
+        SUBJECT_ALT_NAME_EXT,
+        "-subj",
+        subject
+      )
+    }
+
+    private suspend fun signCertificate(name: String, rootCertName: String) {
+      runCommand(
+        "openssl",
+        "x509",
+        "-in",
+        "$name.csr",
+        "-out",
+        "$name.pem",
+        "-req",
+        "-CA",
+        "$rootCertName.pem",
+        "-CAkey",
+        "$rootCertName.key",
+        "-CAcreateserial",
+        "-extfile",
+        "san.ext"
+      )
+    }
```

Comment:
Won't this be nondeterministic? Should we commit pregenerated files? Maybe set notAfter to 99991231235959Z so we don't have to worry about the certs expiring? Include a shell script so that we know how they were generated and can regenerate them if we want later?

Reply:
It does result in new key pairs each time, but isn't nondeterministic in a way that would cause the test to be flaky.

The only change I would consider here is to move the generation into genrule targets so that they get cached across runs (and therefore would need delayed expiration).